### PR TITLE
Add support for excluding dev dependencies from lockfiles.

### PR DESCRIPTION
**Pull request by mariusvniekerk**

### CODE REVIEW

File: conda_lock/src_parser/poetry.py [hunk 1]
```diff
@@ -32,15 +32,19 @@ def join_version_components(pieces):
...
     else:
         return name
 
 
 def poetry_version_to_conda_version(version_string):
     components = [c.replace(" ", "").strip() for c in version_string.split(",")]
     output_components = []
+
     for c in components:
+        if len(c) == 0:
```
Comment (scopatz) [inline conda_lock/src_parser/poetry.py:46]: Not that it matters much here, but I think `not c` is faster than `len(c) == 0`

File: conda_lock/src_parser/poetry.py [hunk 2]
```diff
@@ -64,17 +68,21 @@ def poetry_version_to_conda_version(version_string):
...
-    pyproject_toml: pathlib.Path, platform: str
+    pyproject_toml: pathlib.Path, platform: str, include_dev_dependencies: bool
 ) -> LockSpecification:
     contents = toml.load(pyproject_toml)
     specs: List[str] = []
-    for key in ["dependencies", "dev-dependencies"]:
-        deps = contents.get("tool", {}).get("poetry", {}).get(key, {})
+    dependency_sections = ["dependencies"]
+    if include_dev_dependencies:
+        dependency_sections += ["dev-dependencies"]
```
Comment (scopatz) [inline conda_lock/src_parser/poetry.py:77]: append?

Comment (mariusvniekerk) [reply conda_lock/src_parser/poetry.py:77]:
```suggestion
        dependency_sections.append("dev-dependencies")
```

File: conda_lock/src_parser/poetry.py (hunk 1 above)
Comment (mariusvniekerk) [reply conda_lock/src_parser/poetry.py:46]: but less explicit

Comment (scopatz) [reply conda_lock/src_parser/poetry.py:46]: fair :wink:

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**